src/rust/engine/process_execution/src/remote/streaming.rs

```diff
@@ -145,20 +145,31 @@ impl StreamingCommandRunner {
   // Outputs progress reported by the server and returns the next actionable operation
   // or gRPC status back to the main loop (plus the operation name so the main loop can
   // reconnect).
-  async fn wait_on_operation_stream<S>(&self, mut stream: S) -> (Option<String>, OperationOrStatus)
+  async fn wait_on_operation_stream<S>(
+    &self,
+    mut stream: S,
+    build_id: &str,
+  ) -> (Option<String>, OperationOrStatus)
   where
     S: Stream<Item = Result<bazel_protos::operations::Operation, grpcio::Error>> + Unpin,
   {
     let mut operation_name_opt: Option<String> = None;
 
-    debug!("monitor_operation_stream: monitoring stream");
+    trace!(
+      "wait_on_operation_stream (build_id={}): monitoring stream",
+      build_id
+    );
 
     loop {
-      // TODO: Add deadline timeout.
       match stream.next().await {
         Some(Ok(operation)) => {
-          debug!("wait_on_operation_stream: got operation: {:?}", &operation);
-          operation_name_opt = Some(operation.get_name().to_string());
+          trace!(
+            "wait_on_operation_stream (build_id={}): got operation: {:?}",
+            build_id,
+            &operation
+          );
+          operation_name_opt =
+            Some(operation.get_name().to_string()).filter(|s| !s.trim().is_empty());
```

Comment:
This deserves a comment probably... I'm not sure what this would be doing.

Reply:
Added comment.